public/app/plugins/datasource/prometheus/querybuilder/shared/QueryOptionGroup.tsx

```diff
@@ -2,42 +2,62 @@ import { css } from '@emotion/css';
 import React from 'react';
 import { useToggle } from 'react-use';
 
-import { GrafanaTheme2 } from '@grafana/data';
+import { getValueFormat, GrafanaTheme2 } from '@grafana/data';
 import { Stack } from '@grafana/experimental';
 import { Icon, useStyles2 } from '@grafana/ui';
+import { QueryStats } from 'app/plugins/datasource/loki/components/types';
+import { LokiDatasource } from 'app/plugins/datasource/loki/datasource';
 
 export interface Props {
   title: string;
   collapsedInfo: string[];
   children: React.ReactNode;
+  queryStats?: QueryStats;
+  datasource?: LokiDatasource;
 }
 
-export function QueryOptionGroup({ title, children, collapsedInfo }: Props) {
+export function QueryOptionGroup({ title, children, collapsedInfo, queryStats, datasource }: Props) {
   const [isOpen, toggleOpen] = useToggle(false);
-  const styles = useStyles2(getStyles);
+  const styles = useStyles2(getStyles(queryStats?.bytes));
+
+  const convertUnits = () => {
+    const { text, suffix } = getValueFormat('bytes')(queryStats?.bytes!, 1);
+    return text + suffix;
+  };
 
   return (
-    <Stack gap={0} direction="column">
-      <div className={styles.header} onClick={toggleOpen} title="Click to edit options">
-        <div className={styles.toggle}>
-          <Icon name={isOpen ? 'angle-down' : 'angle-right'} />
-        </div>
-        <h6 className={styles.title}>{title}</h6>
-        {!isOpen && (
-          <div className={styles.description}>
-            {collapsedInfo.map((x, i) => (
-              <span key={i}>{x}</span>
-            ))}
+    <div className={styles.wrapper}>
+      <Stack gap={0} direction="column">
+        <div className={styles.header} onClick={toggleOpen} title="Click to edit options">
+          <div className={styles.toggle}>
+            <Icon name={isOpen ? 'angle-down' : 'angle-right'} />
           </div>
-        )}
-      </div>
-      {isOpen && <div className={styles.body}>{children}</div>}
-    </Stack>
+          <h6 className={styles.title}>{title}</h6>
+          {!isOpen && (
+            <div className={styles.description}>
+              {collapsedInfo.map((x, i) => (
+                <span key={i}>{x}</span>
+              ))}
+            </div>
+          )}
+        </div>
+        {isOpen && <div className={styles.body}>{children}</div>}
+      </Stack>
+      {datasource?.type === 'loki' && queryStats?.bytes ? (
```

Comment:
Rather than having this conditional statement, how do we all feel about taking this opportunity to clone also this component into their separate components for Loki/Prom?
@grafana/observability-logs @grafana/observability-metrics

Reply:
But do we need this data source check here? I'd argue that if we don't have stats, then we shouldn't render it no matter what data source. And if we have stats, then we should render it no matter what data source.